src/main/java/org/terasology/las/yinyang/YinYangProvider.java

```diff
@@ -5,33 +5,54 @@
 import org.joml.Vector3i;
 import org.terasology.engine.world.block.BlockRegion;
 import org.terasology.engine.world.generation.Border3D;
+import org.terasology.engine.world.generation.Facet;
 import org.terasology.engine.world.generation.FacetProviderPlugin;
 import org.terasology.engine.world.generation.GeneratingRegion;
 import org.terasology.engine.world.generation.Produces;
+import org.terasology.engine.world.generation.Updates;
+import org.terasology.engine.world.generation.facets.SurfacesFacet;
 import org.terasology.engine.world.generator.plugin.RegisterPlugin;
+import org.terasology.ligthandshadow.componentsystem.LASUtils;
 
 import java.util.Collections;
 import java.util.List;
 
 @RegisterPlugin
 @Produces(YinYangFacet.class)
+@Updates(@Facet(value = SurfacesFacet.class))
 public class YinYangProvider implements FacetProviderPlugin {
 
-    private List<Vector3i> yinYangPositions = Collections.singletonList(new Vector3i(0, 10, 0));
+    private static final int RADIUS = 5;
+    private final List<Vector3i> yinYangPositions = Collections.singletonList(new Vector3i(0, 10, 0));
 
     @Override
     public void process(GeneratingRegion region) {
         Border3D border =
                 region.getBorderForFacet(YinYangFacet.class)
                         .extendBy(0, 0, 10);
 
+        SurfacesFacet surfacesFacet = region.getRegionFacet(SurfacesFacet.class);
+
         YinYangFacet yinYangFacet = new YinYangFacet(region.getRegion(), border);
 
         BlockRegion worldRect = yinYangFacet.getWorldRegion();
-
+        Vector3i[] tempPos = new Vector3i[1];
+        tempPos[0] = new Vector3i();
```

Comment:
Why do you use an array with a single element here? You could just use the temp vector directly :thinking:

Reply:
https://www.baeldung.com/java-8-lambda-expressions-tips#mutation

Reply:
That was required to make use of mutable variables in a lambda expression.

Reply:
That's not exactly a "best practices" example, but rather a showcase how the "effectively final" rule  is helpful, but not sufficient to prevent accidental errors. 

That said, we luckily have a better option here: Using a similar trick, we'll use a final `Vector3i` (we won't change the reference/object pointer) and only `.set(...)` its value in the lambda expression.

Note that we never do a reassignment `tmp = ...` but only mutation through method calls `tmp.set(...)`. This also helps to keep the number of instantiations low.